cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java

```diff
@@ -178,4 +181,22 @@ public ProgramController run(Program program, ProgramOptions options) {
     }
     return controller;
   }
+
+  @Nullable
```

Comment:
Not sure what use of annotation Nullable here? It is used primarily to indicate if we could pass null as value for input parameters.
Maybe using JavaDoc explaining it could return null instead.